### PR TITLE
CP/add performer to dto

### PR DESCRIPTION
**Pull request by MelnykVL**
Added performer field to TaskDto

### CODE REVIEW

File: cp/app/src/main/java/org/hl7/gravity/refimpl/sdohexchange/service/TaskService.java [hunk 1]
```diff
@@ -49,6 +49,9 @@
 
   public List<TaskDto> readAll() {
     Bundle tasksBundle = taskRepository.findAllTasks();
+    FhirUtil.getFromBundle(tasksBundle, Task.class)
+        .forEach(
+            task -> task.setOwnerTarget(FhirUtil.getFirstFromBundle(taskRepository.findOurTask(task), Task.class)));
```
Comment (yuflyud) [inline cp/app/src/main/java/org/hl7/gravity/refimpl/sdohexchange/service/TaskService.java:54]: If you used _revinclude - the OwnerTarget would be set automatically

Comment (yuflyud) [reply cp/app/src/main/java/org/hl7/gravity/refimpl/sdohexchange/service/TaskService.java:54]: This is super inefficient. You are sending a FHIR request for every record

File: cp/app/src/main/java/org/hl7/gravity/refimpl/sdohexchange/service/TaskService.java [hunk 2]
```diff
@@ -52,6 +53,23 @@
 
   public List<TaskDto> readAll() {
     Bundle tasksBundle = taskRepository.findAllTasks();
+    FhirUtil.getFromBundle(tasksBundle, Task.class)
```
Comment (yuflyud) [inline cp/app/src/main/java/org/hl7/gravity/refimpl/sdohexchange/service/TaskService.java:56]: Move this logic to converter. This is the whole idea of converters - you pass a bundle and it transforms it to dtos. Let it handle order/filler-order tasks itself.

File: components/fhir-common/src/main/java/org/hl7/gravity/refimpl/sdohexchange/dao/FhirRepository.java [hunk 3]
```diff
@@ -56,6 +56,22 @@ public Bundle find(String id, Collection<Include> includes) {
         .execute();
```
Comment (MelnykVL) [inline components/fhir-common/src/main/java/org/hl7/gravity/refimpl/sdohexchange/dao/FhirRepository.java:56]: Call find() with three parameters

File: components/fhir-common/src/main/java/org/hl7/gravity/refimpl/sdohexchange/fhir/extract/BundleExtractor.java [hunk 4]
```diff
@@ -25,6 +29,10 @@
         .collect(Collectors.toList());
   }
 
+  /**
+   * Get map of resources from the bundle
```
Comment (MelnykVL) [inline components/fhir-common/src/main/java/org/hl7/gravity/refimpl/sdohexchange/fhir/extract/BundleExtractor.java:33]: Grouped by the resource type

File: components/fhir-common/src/main/java/org/hl7/gravity/refimpl/sdohexchange/fhir/extract/BundleExtractor.java [hunk 5]
```diff
@@ -25,6 +29,10 @@
...
+   * Usually use it with {@link #resourceList(Map, Class)} method
```
Comment (MelnykVL) [inline components/fhir-common/src/main/java/org/hl7/gravity/refimpl/sdohexchange/fhir/extract/BundleExtractor.java:34]: Remove

File: cp/app/src/main/java/org/hl7/gravity/refimpl/sdohexchange/dao/impl/TaskRepository.java [hunk 6]
```diff
@@ -29,6 +29,8 @@ public Bundle findAllTasks() {
             .code(Task.TaskIntent.ORDER.toCode()))
         // include ServiceRequest
         .include(Task.INCLUDE_FOCUS)
+        .revInclude(Task.INCLUDE_BASED_ON)
+        .include(Task.INCLUDE_OWNER)
```
Comment (MelnykVL) [inline cp/app/src/main/java/org/hl7/gravity/refimpl/sdohexchange/dao/impl/TaskRepository.java:33]: Make like iterate

File: cp/app/src/main/java/org/hl7/gravity/refimpl/sdohexchange/fhir/extract/CpTaskInfoBundleExtractor.java [hunk 7]
```diff
@@ -0,0 +1,76 @@
...
+import java.util.stream.Collectors;
+
+public class CpTaskInfoBundleExtractor extends BundleExtractor<List<CpTaskInfoHolder>> {
+
+  private final TaskInfoBundleExtractor taskInfoBundleExtractor = new TaskInfoBundleExtractor();
+
+  @Override
+  public List<CpTaskInfoHolder> extract(Bundle bundle) {
+    List<TaskInfoHolder> taskInfoHolders = taskInfoBundleExtractor.extract(bundle);
+    Map<String, Task> ourTaskMap = FhirUtil.getFromBundle(bundle, Task.class)
```
Comment (MelnykVL) [inline cp/app/src/main/java/org/hl7/gravity/refimpl/sdohexchange/fhir/extract/CpTaskInfoBundleExtractor.java:24]: Rename to "taskIdToOurTaskMap"

File: cp/app/src/main/java/org/hl7/gravity/refimpl/sdohexchange/fhir/extract/CpTaskInfoBundleExtractor.java [hunk 8]
```diff
@@ -0,0 +1,76 @@
...
+            .getIntent() == Task.TaskIntent.ORDER)
+        .map(taskInfoHolder -> {
+          Task ourTask = ourTaskMap.get(taskInfoHolder.getTask()
+              .getIdElement()
+              .getIdPart());
+          Organization performer = null;
+          if (!Objects.isNull(ourTask)) {
+            if (!(ourTask.getOwner()
+                .getResource() instanceof Organization)) {
+              String reason = String.format("Our task resource with id '%s' does not contain owner.",
```
Comment (MelnykVL) [inline cp/app/src/main/java/org/hl7/gravity/refimpl/sdohexchange/fhir/extract/CpTaskInfoBundleExtractor.java:44]: Add "of type Organization."

File: cp/app/src/main/java/org/hl7/gravity/refimpl/sdohexchange/fhir/extract/OurTaskInfoBundleExtractor.java [hunk 9]
```diff
@@ -0,0 +1,59 @@
...
+    return taskInfoHolders.stream()
+        .filter(t -> t.getTask()
+            .getIntent() == Task.TaskIntent.FILLERORDER)
+        .map(taskInfoHolder -> {
+          Task ourTask = taskInfoHolder.getTask();
+          if (ourTask.getBasedOn()
+              .isEmpty() || !(ourTask.getBasedOn()
+              .get(0)
+              .getResource() instanceof Task)) {
+            String reason = String.format("Our task resource with id '%s' does not contain basedOn.",
```
Comment (MelnykVL) [inline cp/app/src/main/java/org/hl7/gravity/refimpl/sdohexchange/fhir/extract/OurTaskInfoBundleExtractor.java:29]: Add "of type Task"

File: cp/app/src/main/java/org/hl7/gravity/refimpl/sdohexchange/dto/converter/CpTaskBundleToDtoConverter.java [hunk 10]
```diff
@@ -0,0 +1,40 @@
...
+import org.hl7.gravity.refimpl.sdohexchange.fhir.extract.CpTaskInfoBundleExtractor;
+import org.hl7.gravity.refimpl.sdohexchange.util.FhirUtil;
+import org.springframework.core.convert.converter.Converter;
+import org.springframework.util.Assert;
+
+import java.util.List;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+public class CpTaskBundleToDtoConverter implements Converter<Bundle, List<TaskDto>> {
```
Comment (MelnykVL) [inline cp/app/src/main/java/org/hl7/gravity/refimpl/sdohexchange/dto/converter/CpTaskBundleToDtoConverter.java:15]: Rename to "TaskBundleToDtoConverter"

**Pull request merged**